File: app/src/main/java/com/frank/ffmpeg/activity/PushActivity.java
package com.frank.ffmpeg.activity;

import android.os.Bundle;
import android.text.TextUtils;
import android.util.Log;
import android.view.View;
import android.widget.EditText;

import com.frank.ffmpeg.Pusher;
import com.frank.ffmpeg.R;

import java.io.File;

/**
 * 使用ffmpeg推流直播
 * Created by frank on 2018/2/2.
 */
public class PushActivity extends BaseActivity {

    private static final String TAG = PushActivity.class.getSimpleName();
    private static final String FILE_PATH = "storage/emulated/0/hello.flv";
    private static final String LIVE_URL = "rtmp://192.168.1.104/live/stream";

    private EditText edit_file_path;

    private EditText edit_live_url;

    @Override
    int getLayoutId() {
        return R.layout.activity_push;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        hideActionBar();
        initView();
    }

    private void initView() {
        edit_file_path = getView(R.id.edit_file_path);
        edit_live_url = getView(R.id.edit_live_url);
        edit_file_path.setText(FILE_PATH);
        edit_live_url.setText(LIVE_URL);

        initViewsWithClick(R.id.btn_push_stream);
    }

    private void startPushStreaming() {
        //TODO 视频文件格式为flv
        final String filePath = edit_file_path.getText().toString();
        final String liveUrl = edit_live_url.getText().toString();
        Log.i(TAG, "filePath=" + filePath);
        Log.i(TAG, "liveUrl=" + liveUrl);

        if(!TextUtils.isEmpty(filePath) && !TextUtils.isEmpty(filePath)){
            File file = new File(filePath);
            //判断文件是否存在
            if(file.exists()){
                //开启子线程
                new Thread(new Runnable() {
                    @Override
                    public void run() {
                        //开始推流
                        new Pusher().pushStream(filePath, liveUrl);
                    }
                }).start();
            }else {
                showToast(getString(R.string.file_not_found));
            }
        }
    }

    @Override
    void onViewClick(View view) {
        if (view.getId() == R.id.btn_push_stream) {
            startPushStreaming();
        }
    }

    @Override
    void onSelectedFile(String filePath) {

    }
}

File: app/src/main/java/com/frank/ffmpeg/activity/PushActivity.kt
package com.frank.ffmpeg.activity

import android.os.Bundle
import android.text.TextUtils
import android.util.Log
import android.view.View
import android.widget.EditText

import com.frank.ffmpeg.FFmpegPusher
import com.frank.ffmpeg.R

import java.io.File

/**
 * Using FFmpeg to push http-flv stream
 * Created by frank on 2018/2/2.
 */
class PushActivity : BaseActivity() {

    private var editInputPath: EditText? = null

    private var editLiveURL: EditText? = null

    override val layoutId: Int
        get() = R.layout.activity_push

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        hideActionBar()
        initView()
    }

    private fun initView() {
        editInputPath = getView(R.id.edit_file_path)
        editLiveURL   = getView(R.id.edit_live_url)
        editInputPath!!.setText(INPUT_PATH)
        editLiveURL!!.setText(LIVE_URL)

        initViewsWithClick(R.id.btn_push_stream)
    }

    private fun startPushStreaming() {
        val filePath = editInputPath!!.text.toString()
        val liveUrl  = editLiveURL!!.text.toString()

        if (!TextUtils.isEmpty(filePath) && !TextUtils.isEmpty(filePath)) {
            Thread(Runnable {
                FFmpegPusher().pushStream(filePath, liveUrl)
            }).start()
        }
    }

    override fun onViewClick(view: View) {
        if (view.id == R.id.btn_push_stream) {
            startPushStreaming()
        }
    }

    override fun onSelectedFile(filePath: String) {

    }

    companion object {

        // storage/emulated/0/beyond.mp4
        private const val INPUT_PATH = "http://clips.vorwaerts-gmbh.de/big_buck_bunny.mp4"
        private const val LIVE_URL = "rtmp://192.168.17.168/live/stream"
    }
}
